### PR TITLE
7341 Testing GridMap

### PR DESCRIPTION
**Pull request by gam4ik**

### CODE REVIEW

File: test/AutoExpandbleMap/AutoExpandble_map_utest.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#include <gtest/gtest.h>
+
+#include "Map_test.h"
+
+TEST(AutoExpandbleMapBaseTests, Right) {
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/AutoExpandble_map_utest.cpp]: Prepend test's name with "Expand_", e.g. Expand_Ringt and so on.

File: test/AutoExpandbleMap/AutoExpandble_map_utest.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  HeWi param = AutoExpandbleMap_Test({2, 0});
+  bool flag = (param.height == 1)&&(param.width == 4);
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/AutoExpandble_map_utest.cpp]: the "flag" name is too generic, make it more specific. Apply this comment to the tests below.

File: test/AutoExpandbleMap/AutoExpandble_map_utest.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  EXPECT_TRUE(flag);
+  flag = (param.x == 0)&&(param.y == 0);
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/AutoExpandble_map_utest.cpp]: Surround && with spaces

File: test/AutoExpandbleMap/Map_test.h [hunk 4]
```diff
@@ -0,0 +1,45 @@
+#ifndef _MAP_TEST_H
+#define _MAP_TEST_H
+
+#include "../../src/core/maps/grid_map.h"
+#include "../../src/tiny_slam/tiny_grid_cells.h"
+
+struct HeWi {
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: HeWi is not a meaningful name

File: test/AutoExpandbleMap/Map_test.h [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  int height;
+  int width;
+  int x;
+  int y;
+};
+
+GridMap map(std::shared_ptr<GridCellFactory>(new TinyBaseCellFactory));
+
+HeWi AutoExpandbleMap_Test(const DiscretePoint2D& Cell_coor) {
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: The name isn't meaningful. Why is it called "blahblah_Test"?

File: test/AutoExpandbleMap/Map_test.h [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  map.update_cell(Cell_coor, {0.5, 0.5});
+  param.height = map.height();
+  param.width  = map.width();
+  param.x      = map.map_center_x();
+  param.y      = map.map_center_y();
+  
+  return param;
+}
+
+bool CellValue_test() {
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: Think about a name that reflects actions the method performs.

File: test/AutoExpandbleMap/Map_test.h [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  int x0 = map.map_center_x() - 1  - map.width();
+  int y0 = map.map_center_y() - 1 - map.height();
+  int i = 0, j = 0;
+
+  if (map.map_center_x() == 0)
+    x0 = 0;
+  if (map.map_center_y() == 0)
+    y0 = 0;
+
+  for (i = x0; i < map.width(); ++i)
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: braces

File: test/AutoExpandbleMap/Map_test.h [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  param.height = map.height();
+  param.width  = map.width();
+  param.x      = map.map_center_x();
+  param.y      = map.map_center_y();
+  
+  return param;
+}
+
+bool CellValue_test() {
+  int x0 = map.map_center_x() - 1  - map.width();
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: combine with 34 via ternary operator

File: test/AutoExpandbleMap/Map_test.h [hunk 9]
```diff
@@ -25,20 +25,21 @@ HeWi AutoExpandbleMap_Test(const DiscretePoint2D& Cell_coor) {
   return param;
 }
 
-bool CellValue_test() {
-  int x0 = map.map_center_x() - 1  - map.width();
-  int y0 = map.map_center_y() - 1 - map.height();
+bool test_CellValueNotChange() {
+  int x0 = (map.map_center_x() == 0) ? 0 :
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: What does condition mean?

File: test/AutoExpandbleMap/Map_test.h [hunk 10]
```diff
@@ -4,7 +4,7 @@
 #include "../../src/core/maps/grid_map.h"
 #include "../../src/tiny_slam/tiny_grid_cells.h"
 
-struct HeWi {
+struct MapParam {
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: MapParam**s**

File: test/AutoExpandbleMap/Map_test.h [hunk 11]
```diff
@@ -4,7 +4,7 @@
...
   int height;
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: int height, width;
int x, y;

What does x mean?

File: test/AutoExpandbleMap/Map_test.h [hunk 12]
```diff
@@ -13,8 +13,8 @@ struct HeWi {
 
 GridMap map(std::shared_ptr<GridCellFactory>(new TinyBaseCellFactory));
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: Can we get rid of the global map variable?

File: test/AutoExpandbleMap/AutoExpandble_map_utest.cpp [hunk 13]
```diff
@@ -2,81 +2,81 @@
...
 #include "Map_test.h"
 
-TEST(AutoExpandbleMapBaseTests, Right) {
-  HeWi param = AutoExpandbleMap_Test({2, 0});
-  bool flag = (param.height == 1)&&(param.width == 4);
-  EXPECT_TRUE(flag);
-  flag = (param.x == 0)&&(param.y == 0);
-  EXPECT_TRUE(flag);
+TEST(AutoExpandbleMapBaseTests, Expand_Right) {
+  MapParam param = test_auto_expand({2, 0});
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/AutoExpandble_map_utest.cpp]: Have you tried to write a macro/function for the expansion verification code?
E.g. test_expansion(point, expected map params).
Have you also considered implementation of operator== for MapParams?

**Review by HatlessFox — CHANGES_REQUESTED** [review]
Minor fixes and clarifications are required.

File: test/AutoExpandbleMap/Map_test.h [hunk 14]
```diff
@@ -4,31 +4,44 @@
...
 
-struct MapParam {
-  int height;
-  int width;
-  int x;
-  int y;
+class MapParams {
+public:
+  MapParams (int height = 0, int width = 0, int map_center_x = 0, int map_center_y = 0) :
+             _height(height), _width(width), _map_center_x(map_center_x),
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: move  both _map_center_x/y to a new line

File: test/AutoExpandbleMap/Map_test.h [hunk 15]
```diff
@@ -4,31 +4,44 @@
...
+
+  int _height;
+  int _width;
+  int _map_center_x;
+  int _map_center_y;
 };
 
-GridMap map(std::shared_ptr<GridCellFactory>(new TinyBaseCellFactory));
+bool operator==(const MapParams &left, const MapParams &right) {
+  return (left._height == right._height)&&(left._map_center_x == right._map_center_x)
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: ) && ( (spaces)

File: test/AutoExpandbleMap/Map_test.h [hunk 16]
```diff
@@ -4,31 +4,44 @@
...
+  param._map_center_x = map.map_center_x();
+  param._map_center_y = map.map_center_y();
+
   return param;
 }
 
-bool test_CellValueNotChange() {
-  int x0 = (map.map_center_x() == 0) ? 0 :
+bool test_CellValueNotChange(GridMap map) {
+  bool is_centershifting_x = (map.map_center_x() == 0),
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: Mb center_x_is_shifted?

File: test/AutoExpandbleMap/AutoExpandble_map_utest.cpp [hunk 17]
```diff
@@ -2,81 +2,53 @@
 
 #include "Map_test.h"
 
+bool is_succeed(const DiscretePoint2D& Cell_coor, MapParams expected_param) {
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/AutoExpandble_map_utest.cpp]: The name is not descriptive. Mb test_expansion_by_cell_update? And you can rm test_auto_expand (it seems redundant).

File: test/AutoExpandbleMap/Map_test.h [hunk 18]
```diff
@@ -4,31 +4,44 @@
...
+MapParams test_auto_expand(const DiscretePoint2D& Cell_coor, GridMap map) {
+  MapParams param;  
 
   map.update_cell(Cell_coor, {0.5, 0.5});
-  param.height = map.height();
-  param.width  = map.width();
-  param.x      = map.map_center_x();
-  param.y      = map.map_center_y();
-  
+  param._height = map.height();
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/Map_test.h]: Use constructor, move the implementation to "is_succeed"

**Comment by HatlessFox** [comment]
Comments were updated.

**Review by HatlessFox — APPROVED** [review]

File: test/AutoExpandbleMap/AutoExpandble_map_utest.cpp [hunk 19]
```diff
@@ -2,46 +2,49 @@
 
 #include "Map_test.h"
 
-bool is_succeed(const DiscretePoint2D& Cell_coor, MapParams expected_param) {
+bool test_expansion_by_cell_update(const DiscretePoint2D& Cell_coor, MapParams expected_param) {
```
Comment (HatlessFox) [inline test/AutoExpandbleMap/AutoExpandble_map_utest.cpp:5]: The line has 80+ length. Split it and check the others.
Cell_coor -> cell_coord

Comment (krinkin) [reply test/AutoExpandbleMap/AutoExpandble_map_utest.cpp:5]: Guys, i think we can increase line length limit up to 100. I mean {80-100] is OK

**Pull request merged**